.github/workflows/test-build-deploy.yml

```diff
@@ -32,6 +32,24 @@ jobs:
         run: |
           mkdir -p /go/src/github.com/grafana/mimir
           ln -s $GITHUB_WORKSPACE/* /go/src/github.com/grafana/mimir
+      - name: Get golangci-lint cache path
+        id: golangcilintcache
+        run: |
+          echo "path=$(golangci-lint cache status | grep 'Dir: ' | cut -d ' ' -f2)" >> "$GITHUB_OUTPUT"
+      - name: Cache golangci-lint cache
+        uses: actions/cache@v3
+        with:
+          key: lint-golangci-lint-${{ runner.os }}-${{ hashFiles('**/go.sum', '.golangci.yml', 'Makefile') }}
+          path: ${{ steps.golangcilintcache.outputs.path }}
+      - name: Get Go build cache path
+        id: gocache
+        run: |
+          echo "path=$(go env GOCACHE)" >> "$GITHUB_OUTPUT"
+      - name: Cache Go build cache
```

Comment:
Can't you use caching built into actions/setup-go? I just merged [the bump to v4](https://github.com/grafana/mimir/pull/5142), which enables caching by default.

Reply:
We only use `setup-go` for the integration test job, and so in the interests of consistency, I went with using `actions/cache` directly everywhere. I don't think we'd want to use `setup-go` for the jobs that run in a container, as we want the container to control the version of Go (although they _should_ always be the same).

Reply:
Although if `setup-go` enables caching by default, we probably want to just use one or the other for caching in the integration test job.